src/schema/v2/partner.ts

```diff
@@ -150,6 +165,24 @@ export const PartnerType = new GraphQLObjectType<any, ResolverContext>({
             gravityArgs.artwork_id = flatten([args.artworkIDs])
           }
 
+          if (args.shallow === false && partnerArtworksAllLoader) {
```

Comment:
Wanted to get your feedback on ways to handle the case when a non partner user is trying to hit this code path. We added this check to the `if` statement above to fix the type error we were running into yesterday during mobbing - the fact that partnerArtworksAllLoader can be undefined (`Cannot invoke an object which is possibly 'undefined'`). Not entirely sure when it would be undefined, we had the theory that it would be undefined if you were not an authenticated partner/admin.


Right now if you are not an authorized partner/admin you'll get a 403 error, we were thinking maybe we should direct those users to the public `partnerArtworksLoader` as they were not getting an 403 when trying to fetch info from here previous to our shallow changes.

Reply:
To answer a few of the questions:
- Correct, if you query this endpoint without a user access token, this loader will be `undefined`. Might help to add a comment specifying that you're adding the check to guard against that case.
- Re: requiring partner-or-admin-level authorization:
  - you could handle it gracefully in code, as we do when [fetching for a causality jwt](https://github.com/artsy/metaphysics/blob/master/src/schema/v1/causality_jwt.ts#L104-L109).
  - This ☝️ might be a little heavy-handed for our use-case. Some other options are to `catch` that particular error and use the loader that doesn't require authorization if it's a 403, and/or to add a comment (or in the description) to the `shallow` param indicating that it only works for partners or admins.

Reply:
Thanks Sarah, that makes a lot of sense. 

The authenticated loader is `undefined` if the requester is not passing in an access token. I think we were trying to test for regular old user making a request here and was surprised to see the loader be present! I can adjust and make the comments/docs more explicit, we shouldnt be affecting the current behavior if `shallow` is not present